mobility_games/games/wav_musical_cane.py

```diff
@@ -0,0 +1,253 @@
+                                                newitem.pose.orientation.z,
+                                                newitem.pose.orientation.w])
+                self.orientation_list.append(angles)
+                self.id_list.append(tag_id)
+        except:
+            for item in msg.detections:
+                x = None
+                y = None
+                z = None
+                tag_id = item.id
+                self.tag_list.append((x, y, z, tag_id))
+                self.id_list.append(tag_id)
+
+
+    def config_callback(self, config, level):
+        self.mode = config["mode"]
+        self.sweep_tolerance = config["sweep_tolerance"]
+        self.sweep_range = config["sweep_range"]
+        return config
+
+    def run_wav(self):
+        """ Plays a wav file while the user is sweeping the cane consistently.
+            Pauses music when the cane stops, and resumes when it continues. """
+
+        r = rospy.Rate(10)
+        print("Searching for Tango...")
+        print("Searching for AR tags...")
+        visible_tag = 0
+
+        while not rospy.is_shutdown():
+
+            if not self.start:
+                print("X position: " + str(self.x),
+                    "Cane x position: " + str(self.cane_x))
+
+            #   Start program if receiving pose from Tango
+            if self.x and self.cane_x and not self.start:
+
+                #   Note initial position
+                self.x_init = self.x
+                self.y_init = self.y
+
+                #   Set some initial conditions
+                self.start = True
+                self.cane_y_prev = self.cane_y
+                last_sweep = rospy.Time.now()
+                tag = 0
+
+                #   Keep track of which tag's audio should be playing and is
+                #   currently playing. -1 means no tags are applicable.
+                should_play = -1
+                playing = -1
+
+                print("Connection established.")
+                print("Cane found.")
+
+                #   Set parameters for cane sweeping.
+                sweep_width = self.sweep_range
+                cane_points = [-sweep_width/2.0, sweep_width/2.0]
+                offset_selection = 0
+
+            if self.start:
+
+                #   Selects the first tag in view
+                if len(self.id_list):
+                    if self.id_list[0] in self.tag_to_music_object:
+                        visible_tag = self.id_list[0]
+
+                #   Determine the next point the cane should reach
+                offset = cane_points[offset_selection]
+
+                #   If the cane has passed the midline (y = 0), start music
+                if (self.cane_y - offset) * (self.cane_y_prev - offset) <= 0:
+                    should_play = visible_tag
+                    last_sweep = rospy.Time.now()
+                    print("Reached point %s." % offset_selection)
+                    offset_selection += 1
```

Comment:
Changed to `offset_selection = (offset_selection + 1) % len(cane_points):`

Left modular so that `cane_points` can be changed to length other than two.